### ISSUE TITLE
Dev requirements appear to be included in galaxy virtualenv

### ISSUE THREAD

**Issue by nuwang** [CloudVE/galaxy-kube-playbook#3]
When installing from a tarball, dev_wheels are always installed (https://github.com/galaxyproject/galaxy/pull/5491#pullrequestreview-95160319). However, installing all dev packages increases the size of the virtualenv significantly, and consequently, any docker images that are built.

In addition, should we try to skip dev wheels, the same comment above implies that we would then not be able to build a production client, since dev wheels are required to build the client. Even though release tarballs do have a per-built client, I guess it's a dev build, not a production build. Therefore, what we need is a way to build a production client but remove dev wheels afterwards. What's the best way to achieve this?

If we can get rid of dev_wheels, we have a decent chance to get ~100MB compressed.

**Comment by jdavcs** [comment]
This is due to the referenced issue ([galaxyproject/galaxy#5491 (review)](https://github.com/galaxyproject/galaxy/pull/5491#pullrequestreview-95160319)).

When a git repo is not found, dev wheels will be set to 1, because (as per discussion in #5441) we assume we need to build the client the first time.  However, dev wheels contain  A LOT MORE than what is needed to build the client. Only yarn and node are necessary (as far as I can tell), which can be easily installed and then removed. Installing all of the dev wheels, on the other hand, seems simply unnecessary. My estimate for the dev wheels is 48M (based on the difference in installed files' size).

I have submitted a PR ([galaxyproject/galaxy#7371](https://github.com/galaxyproject/galaxy/pull/7371)), which reverses the previous #5491. However, regardless of this PR, this issue seems to have been solved by excluding common_startup.sh from Dockerfile (41d66fd).
